sdk/pthread/pthread_tls.cpp

```diff
@@ -173,7 +174,13 @@ void _pthread_tls_destructors(void)
     struct sgx_pthread_storage *rs;
     int i;
 
-    if(NULL == pthread_info_tls.m_pthread)
+    // For a child thread created inside enclave, we need to clear the TLS when
+    // the thread exits.
+    // In other cases, we need to make decision based on the tcs policy:
+    //   1) Unbinding mode, we need to clear the TLS before the root ecall exits.
+    //   2) Binding mode, we need to persist the TLS across multiple ECALLs. In this
+    //      case, the destructor should not be called.
+    if(NULL == pthread_info_tls.m_pthread && true == is_tcs_binding_mode())
```

Comment:
Then that's the same with my fix. That's been my point since the beginning. -_-

Reply:
As I mentioned in the comments, for the child threads that created via enclave's pthread_create(), we need to always call the destructors before ecall exits.  Considering the case that the enclave is signed with unbinding mode and the child threads use pthread TLS. If we use the check above, then there will be potential memory leak in child threads.

Reply:
Yes but `m_local_storage` will also allow the destructor to be called if in case the `m_pthread` is not `NULL`. As I have been pointed out before, everything inside that function does not depend on the `m_pthread` value. If the child thread created from enclave's `pthread_create` exits, and it has `m_local_storage` not null,  the destructor will always still be called, since it is called by the `do_ecall` function in the tRTS, regardless whether it is a thread created inside the enclave or outside.

Reply:
Yes. I think that's the correct logic. We need to call destructor for the child threads created inside enclave regardless enclave's configuration.

Just want to double confirm the comment at the beginning. Do you want to update the check from 
```
    if(NULL == pthread_info_tls.m_pthread && true == is_tcs_binding_mode())
        //do nothing
        return;
```
To 
```
if (NULL == pthread_info_tls.m_local_storage || true == is_tcs_binding_mode())
      return;
.......
```
If that's the case, I think it is not correct as I explained in my previous reply.

Reply:
Ah ok, I think this commit's condition is logically equivalent to:

if (pthread_info_tls.m_pthread) { // pthread_create inside enclave
  // always cleanup for thread created inside enclave regardless of binding
} else {
  if (is_tcs_binding_mode() {
    // do not destroy and persist?
  } else {
    // destroy?
  }
}

ignore my suggestion.